### DISCUSSION TITLE
Paging not working

### DISCUSSION THREAD

**Discussion by umerkle** [dsbenghe/Novell.Directory.Ldap.NETStandard#243] [General]
Hi!,

i am trying to get 10000 records from an OpenLDAP directory.
I am using Novell.Directory.Ldap.NETStandard NuGet package 3.6.0.

When i try to use SearchWithSimplePaging i get Limit Exceeded Errors.
So i tried to use LdapVirtualListControl on my own (like here: https://stackoverflow.com/questions/48805577/how-to-do-a-paged-search-on-an-ldap-server-with-lots-of-users). 
But i get 'Unavailable Critical Extension (12)'.

What am i doing wrong?

I create a SimplePagedResultsControleHandler with
`
var handler = SimplePagedResultsControleHandler(connection);
pagedResults = handler.SearchWithSimplePaging(GetSearchOptions(searchScope, attributeOnly, attrs, searchBase, searchFilter, null), pageSize);
`

**Comment by umerkle** [comment]
Additional comment:
When i check the requests with Wireshark, i can see that when using SimplePagedResultsControlHandler it tries to page and it even does this correct for some pages (number differs depending on page size). But after some pages it gets Size Limit Exceeded (4).
I tried with different page sizes.
When i look at the traffic from a LDAP tool, it does exactly the same but without getting the error????

**Comment by dsbenghe** [comment]
You likely get _Limit Exceeded_ because you are using a too large page size relative with what is configured in your ldap server
_Unavailable Critical Extension_ means either your ldap server doesn't support VLV or VLV extension is not configured.

**Reply by umerkle** [reply]
I also thought about VLV. But if the cause would be on the LDAP Server, every tool should have the same problem. As i wrote: i checked traffic from said LDAP viewer and traffic from my code. No difference, except the result. So i suspected that the Novell library is the source of the problem.

But besides that, i was able to strip down my queries, to keep the resultset small enough.
